models/image_segmentation_efficientsam/demo.py

```diff
@@ -0,0 +1,137 @@
+    return vis_result
+
+def select(event, x, y, flags, param):
+    global clicked_left
+    # When the left mouse button is pressed, record the coordinates of the point where it is pressed
+    if event == cv.EVENT_LBUTTONUP:
+        point.append([x,y])
+        print("point:",point[0])
+        clicked_left = True
+
+if __name__ == '__main__':
+    # Load the EfficientSAM model
+    model = EfficientSam(modelPath=args.model)
+
+    if args.input is not None:
+        # Read image
+        image = cv.imread(args.input)
+        if image is None:
+            print('Could not open or find the image:', args.input)
+            exit(0)
+        # create window
+        image_window = "image: click on the thing whick you want to segment!"
+        cv.namedWindow(image_window, cv.WINDOW_NORMAL)
+        # change window size
+        cv.resizeWindow(image_window, 800 if image.shape[0] > 800 else image.shape[0], 600 if image.shape[1] > 600 else image.shape[1])
+        # put the window on the left of the screen
+        cv.moveWindow(image_window, 50, 100)
+        # set listener to record user's click point
+        cv.setMouseCallback(image_window, select)
+        # tips in the terminal
+        print("click the picture on the LEFT and see the result on the RIGHT!")
+        # show image
+        cv.imshow(image_window, image)
+        # waiting for click
+        while cv.waitKey(1) == -1 or clicked_left:
+            # receive click
+            if clicked_left:
+                # put the click point (x,y) into the model to predict
+                result = model.infer(image=image, points=point, lables=[1])
+                # get the visualized result
+                vis_result = visualize(image, result)
+                # create window to show visualized result
+                cv.namedWindow("vis_result", cv.WINDOW_NORMAL)
+                cv.resizeWindow("vis_result", 800 if vis_result.shape[0] > 800 else vis_result.shape[0], 600 if vis_result.shape[1] > 600 else vis_result.shape[1])
+                cv.moveWindow("vis_result", 851, 100)
+                cv.imshow("vis_result", vis_result)
+                # set click false to listen another click
+                clicked_left = False
+            elif cv.getWindowProperty(image_window, cv.WND_PROP_VISIBLE) < 1: 
+                # if click × to close the image window then ending
+                break
+            else:
+                # when not clicked, set point to empty
+                point = []
+        cv.destroyAllWindows()
+        
+        # Save results if save is true
+        if args.save:
+            cv.imwrite('./example_outputs/vis_result.jpg', vis_result)
+            cv.imwrite("./example_outputs/mask.jpg", result)
+            print('vis_result.jpg and mask.jpg are saved to ./example_outputs/')
+
+        
+    else:
+        pass
+        '''
+        since the model need about 2s to predict, the camera demo couldn't support now, I will try to update later
```

Comment:
Camera does not make a lot sense regardless inference time. Camere stream gives a sequence of images and we just do not stop at each frame, click on the object, perform segmentation at each frame.

Instead, it would be nice to have a function which loads all images in a directory and perform encoding first, save intermediates. After all images are encoded, we can load the intermediates then click and segment. So in total we can offer two pipeline:

```
One image -> Demo(encoding, segmentation) -> Result
Multiple images -> Demo(encoding, save templates) -> Demo(load templates, segmentation) -> Results
```

Reply:
I propose to remove camera branch for now.